rpc/dial.go

```diff
@@ -329,7 +327,7 @@ func dialMulticastDNS(
 
 	// IPv6 with scope does not work with grpc-go which we would want here.
 	if !(hasGRPC || hasWebRTC) || len(entry.AddrIPv4) == 0 {
-		return nil, false, nil
+		return nil, false, errors.New("mDNS query failed to find an IPv4 candidate")
```

Comment:
I talked with Ethan a bit offline; rust-utils has a slightly different architecture (in particular, I don't think parallel mDNS/WebRTC dials have been implemented yet) for how it returns mDNS connections, but [in the analogous piece of code](https://github.com/viamrobotics/rust-utils/blob/main/src/rpc/dial.rs#L364-L371), it returns `None` and doesn't really report an error. I think returning an error here is ultimately just more, helpful information, but if we do this, we should probably try to get the same behavior in rust.

Reply:
gotcha, thanks for that data point... I agree we should make these behave the same.

my general understanding is that the rest of the dial logic in goutils does not support ipv6 addresses, so we might as well return an error (and maybe export that error) - otherwise downstream code becomes more complicated since it has to handle "no error but also no connection cases".

Reply:
actually, I guess the error description isn't quite comprehensive - we also return `nil, nil` when the candidate we find has neither grpc nor webrtc as a text field

Reply:
ya, think we should update the error to say `failed to find an IPv4 candidate or the candidate we found has neither grpc nor webrtc as a text field`

Reply:
added a custom error for this case: https://github.com/viamrobotics/goutils/pull/270/commits/f2020141d3d02b6365cdd7a8762089e0a23ccbb1